src/Array.jl

```diff
@@ -6,205 +6,163 @@ using LinearAlgebra: Eigen, LinearAlgebra, qr
-        -1im*cis(-g.θ / 2 - g.ϕ)*sin(g.θ / 2) cis(g.θ / 2)*cos(g.θ / 2)
+        cis(op.θ / 2)*cos(op.θ / 2) -1im*cis(-op.θ / 2 + op.ϕ)*sin(op.θ / 2)
+        -1im*cis(-op.θ / 2 - op.ϕ)*sin(op.θ / 2) cis(op.θ / 2)*cos(op.θ / 2)
     ],
 )
 
-Matrix{T}(g::G) where {T,G<:Gate{FSim}} = Matrix{T}([
+Matrix{T}(op::FSim) where {T} = Matrix{T}([
     1 0 0 0
-    0 cos(g.θ) -1im*sin(g.θ) 0
-    0 -1im*sin(g.θ) cos(g.θ) 0
-    0 0 0 cis(-g.ϕ)
+    0 cos(op.θ) -1im*sin(op.ϕ) 0
+    0 -1im*sin(op.ϕ) cos(op.θ) 0
+    0 0 0 1
 ])
 
-function Matrix{T}(::Type{Gate{Op}}) where {T,Op<:Control}
-    n = length(Op)
+function Matrix{T}(op::Op) where {T,Op<:Control}
+    N = length(Op)
     t = length(targettype(Op))
 
-    M = Matrix{T}(LinearAlgebra.I, 2^n, 2^n)
-
-    M[(2^n-2^t+1):end, (2^n-2^t+1):end] = Matrix{T}(Gate{targettype(Op)})
-
-    return M
-end
-
-function Matrix{T}(g::Gate{<:Control}) where {T}
-    n = (length ∘ lanes)(g)
-    t = (length ∘ target)(g)
-
-    M = Matrix{T}(LinearAlgebra.I, 2^n, 2^n)
+    M = Matrix{T}(LinearAlgebra.I, 2^N, 2^N)
 
-    M[(2^n-2^t+1):end, (2^n-2^t+1):end] = Matrix{T}(Gate{targettype(operator(g))}(target(g)...; parameters(g)...))
+    M[(2^N-2^t+1):end, (2^N-2^t+1):end] = Matrix{T}(target(op))
 
     return M
 end
 
-function Base.rand(::Type{SU{N}}, lanes::NTuple{M,Int}; eltype::Type = ComplexF64) where {N,M}
-    # keep unitary matrix Q from QR decomposition
-    q, _ = qr(rand(eltype, N, N))
-
-    SU{N}(lanes...; array = Matrix(q))
+function Matrix{T}(op::SU) where {T}
+    return Matrix{T}(op.matrix)
 end
 
-Base.rand(::Type{Gate{SU{N}}}, lanes::Integer...; kwargs...) where {N} = rand(SU{N}, lanes; kwargs...)
+Array(x::Operator) = Array{ComplexF64}(x)
 
-function Matrix{T}(g::Gate{<:SU{N}}) where {T,N}
-    return g.array |> Matrix{T}
-end
-
-Array(x::Gate) = Array{ComplexF32}(x)
-Array(::Type{T}) where {T<:Gate} = Array{ComplexF32}(T)
-
-for Op in [I, X, Y, Z, H, S, Sd, T, Td, Swap]
-    @eval Array{T}(::G) where {T,G<:Gate{$Op}} = Array{T}(G)
-end
-
-Array{T}(::Type{G}) where {T,G<:Gate} = Array{T,2 * length(operator(G))}(G)
-Array{T}(g::G) where {T,G<:Gate} = Array{T,2 * length(operator(G))}(isparametric(operator(G)) ? g : G)
+Array{T}(op::Op) where {T,Op<:Operator} = Array{T,2 * length(Op)}(op)
 
 # NOTE multidimensional `Array` literal concatenation was introduced in 1.7
 # TODO clean code when we stop supporting Julia 1.6
-Array{T,4}(::Type{<:Gate{Swap}}) where {T} = Array{T}([1; 0;; 0; 0;;; 0; 0;; 1; 0;;;; 0; 1;; 0; 0;;; 0; 0;; 0; 1])
+Array{T,4}(::Swap) where {T} = Array{T}([1; 0;; 0; 0;;; 0; 0;; 1; 0;;;; 0; 1;; 0; 0;;; 0; 0;; 0; 1])
 
-Array{T,4}(g::G) where {T,G<:Gate{FSim}} =
-    Array{T}([1; 0;; 0; 0;;; 0; cos(g.θ);; -1im*sin(g.ϕ); 0;;;; 0; -1im*sin(g.ϕ);; cos(g.θ); 0;;; 0; 0;; 0; 1])
+Array{T,4}(op::FSim) where {T} = Array{T}(
+    [1; 0;; 0; 0;;; 0; cos(op.θ);; -1im*sin(op.θ); 0;;;; 0; -1im*sin(op.θ);; cos(op.θ); 0;;; 0; 0;; 0; cis(-op.θ)],
```

Comment:
Like the other comment, the last element from `FSim` should be:
The last row should be $`e^{-i\phi}`$
You used $\theta$ as the phase instead of $\phi$